src/masternode.cpp

```diff
@@ -200,65 +203,59 @@ void CMasternode::Check(bool forceCheck)
     activeState = MASTERNODE_ENABLED; // OK
 }
 
-int64_t CMasternode::SecondsSincePayment() {
-    int64_t sec = (GetAdjustedTime() - GetLastPaid());
-    int64_t month = 60*60*24*30;
-    if(sec < month) return sec; //if it's less than 30 days, give seconds
-
-    CHashWriter ss(SER_GETHASH, PROTOCOL_VERSION);
-    ss << vin;
-    ss << sigTime;
-    uint256 hash =  ss.GetHash();
-
-    // return some deterministic value for unknown/unpaid but force it to be more than 30 days old
-    return month + UintToArith256(hash).GetCompact(false);
-}
+int CMasternode::GetCollateralAge()
+{
+    if (chainActive.Height() < 0) return -1;
```

Comment:
Needs a LOCK(cs_main) (unless all callers are guaranteed to have it locked already).